webpack/assets/javascripts/react_app/components/PF4/TableIndexPage/TableIndexPage.js

```diff
@@ -0,0 +1,251 @@
+import React, { useState, useMemo, useEffect, useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { QuestionCircleIcon } from '@patternfly/react-icons';
+import { useHistory } from 'react-router-dom';
+import URI from 'urijs';
+
+import {
+  Spinner,
+  Toolbar,
+  ToolbarContent,
+  ToolbarGroup,
+  ToolbarItem,
+  PageSection,
+  PageSectionVariants,
+  TextContent,
+  Text,
+} from '@patternfly/react-core';
+import {
+  createURL,
+  exportURL,
+  helpURL,
+  changeQuery,
+  getURIsearch,
+} from '../../../common/urlHelpers';
+import { translate as __ } from '../../../common/I18n';
+
+import { useAPI } from '../../../common/hooks/API/APIHooks';
+import { getControllerSearchProps, STATUS } from '../../../constants';
+import { getResults } from '../../AutoComplete/AutoCompleteActions';
+import { TRIGGERS } from '../../AutoComplete/AutoCompleteConstants';
+import BreadcrumbBar from '../../BreadcrumbBar';
+import SearchBar from '../../SearchBar';
+import Head from '../../Head';
+import { ActionButtons } from './ActionButtons';
+import './TableIndexPage.scss';
+
+const TableIndexPage = ({
+  apiUrl,
+  apiOptions,
+  header,
+  breadcrumbOptions,
+  beforeToolbarComponent,
+  controller,
+  searchable,
+  exportable,
+  creatable,
+  hasHelpPage,
+  customSearchProps,
+  customExportURL,
+  customCreateURL,
+  customHelpURL,
+  customActionButtons,
+  cutsomToolbarItems,
+  children,
+}) => {
+  const [params, setParams] = useState({});
+  // Adding to the deps apiOptions creates an endless loop
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const memoOptions = useMemo(() => ({ ...apiOptions, params }), [params]);
+  const history = useHistory();
+  const { location: { search } = {} } = history || {};
+  const {
+    response: { search: apiSearchQuery, can_create: canCreate },
+    status = STATUS.PENDING,
+  } = useAPI('get', apiUrl, memoOptions);
+
+  const dispatch = useDispatch();
+  const memoDefaultSearchProps = useMemo(
+    () => getControllerSearchProps(controller),
+    [controller]
+  );
+  const searchProps = customSearchProps || memoDefaultSearchProps;
+
+  const setAutocomplete = useCallback(
+    newSearch => {
+      dispatch(
+        getResults({
+          url: searchProps.autocomplete.url,
+          searchQuery: newSearch,
+          controller: searchProps.controller,
+          trigger: TRIGGERS.INPUT_CHANGE,
+          id: searchProps.autocomplete.id,
+        })
+      );
+    },
+    [searchProps, dispatch]
+  );
+  useEffect(() => {
+    let newSearch;
+    if (search !== undefined) {
+      const urlParams = new URLSearchParams(search);
+      newSearch = urlParams.get('search');
+    } else {
+      newSearch = getURIsearch();
+    }
+    setAutocomplete(newSearch || '');
```

Comment:
why do we need to trigger the autocomplete#getResults here? shouldn't it be handled in the autocomplete component?

Reply:
Gonna address it here:
https://projects.theforeman.org/issues/34546

Reply:
^ was resolved so `setAutocomplete` is deleted